### PR TITLE
Delete IDE dispose analyzers

### LINKED ISSUE #36507: [QUESTION] Should IDE0067 be disabled in UnitTest ?
**Version Used**:
`global.json` > `2.2.300`
`<LangVersion>latest</LangVersion>`
Vs2019 16.2.0

**Steps to Reproduce**:

1. `dotnet new xunit`
2. Add a Unit test method
3. `var foo = new HttpRequestMessage`
4. Should i be warned about it ?

I do have a `Directory.Build.props` in the `test` folder so i guess there is a way to change that behavior ?

**Expected Behavior**:
Not sure but since unit test are short lived in their own `dotnet test` execution, should this be raised ?
But is that intended to affect unit test ? It's going to add "noise" in the test when someone go read them.
I don't mind adding it for good reasons but since unit test are short lived / process destroyed right after, i'm really surprised

### LINKED ISSUE #38204: IDE0067 fires for IHttpClientFactory.CreateClient(), but not IHttpClientFactory.CreateClient("foo")
**Version Used**: 
Visual Studio Enterprise 16.2.3

**Steps to Reproduce**:
Add the following method in your project:
```csharp
public void DoSomething(IHttpClientFactory clientFactory)
{
    var clientWithWarning = clientFactory.CreateClient();
    clientWithWarning.BaseAddress = new Uri("https://localhost");

    var clientWithoutWarning = clientFactory.CreateClient("Foo");
    clientWithoutWarning.BaseAddress = new Uri("https://localhost");
}
```

**Expected Behavior**:
Get either an IDE0067 warning for both methods or none.

**Actual Behavior**:
The first method without parameters produces a warning message IDE0067, the second doesn't.

### LINKED ISSUE #38900: [IDE0069] Disposable fields when passed from the outside should not be disposed
**Version Used**: 
Visual Studio 16.3.0

**Steps to Reproduce**:

According to [CA2213](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2213-disposable-fields-should-be-disposed?view=vs-2019) only disposable fields that were created locally should be disposed inside the `Dispose` method. But **If the object is created or assigned outside of type T, the rule does not fire. This reduces noise for cases where the containing type doesn't own the responsibility for disposing of the object.**.

It means that the following code should not trigger IDE0069:

```csharp
internal class FooBar : IDisposable
{
    private readonly Stream _d;

    public FooBar(Stream d)
    {
        _d = d;
    }

    public void Dispose()
    {
        throw new NotImplementedException();
    }

    public override string ToString()
    {
        return _d.ToString();
    }
}
```

But it does.

**Expected Behavior**:

`FooBar` should not have any IDE0069 diagnostics.

**Actual Behavior**:

The IDE0069 diagnostic is emitted for disposable fields passed via constructor

### LINKED ISSUE #38944: Erroneous suggestion IDE0067 for disposable object used in 'await using (...) { }'
**Version Used**:
Version 16.4.0 Preview 1.0
VisualStudio.16.Preview/16.4.0-pre.1.0+29319.158

**Steps to Reproduce**:
Paste the following code into a project:
```cs
using System;
using System.IO;
using System.Threading.Tasks;

static class DisposingAnalysis
{
    static void ImplicitSync(ReadOnlySpan<byte> bytes)
    {
        //No suggestions, expected
        using var stream = new MemoryStream();
        stream.Write(bytes);
    }

    static async Task ImplicitAsync(ReadOnlyMemory<byte> bytes)
    {
        //No suggestions, expected
        await using var stream = new MemoryStream();
        await stream.WriteAsync(bytes);
    }

    static void ExplicitSync(ReadOnlySpan<byte> bytes)
    {
        //IDE0063: 'using' statement can be simplified, expected
        using (var stream = new MemoryStream())
        {
            stream.Write(bytes);
        }
    }

    static async Task ExplicitAsync(ReadOnlyMemory<byte> bytes)
    {
        //IDE0063 *AND* IDE0067: Disposable object created by 'new MemoryStream()' is never disposed
        await using (var stream = new MemoryStream())
        {
            await stream.WriteAsync(bytes);
        }
    }
}
```

**Expected Behavior**:
Only IDE0063 is reported for both of the explicit syntax methods to point out the new Implictly Scoped `using` Statement syntax.

**Actual Behavior**:
The combination of the older, explicitly-scoped syntax and async disposable makes the compiler/IDE think the object will not get disposed and reports IDE0067 in addition to the expected behavior.

### LINKED ISSUE #38984: Re-enable IDE Dispose analyzers by default
We are going to disable the DFA based IDE dispose analyzers (IDE0067, IDE0068 and IDE0069) by default as they need more performance tuning to handle large method bodies (method bodies which are 1000s of lines of code) with large control flow graphs.

Meanwhile, analyzers can be enabled by using the following editorconfig entries:
```
[*.{cs,vb}]
dotnet_diagnostic.IDE0067.severity = warning
dotnet_diagnostic.IDE0068.severity = warning
dotnet_diagnostic.IDE0069.severity = warning
```

### PR DESCRIPTION
**Pull request by mavasani**
- These rules have been disabled by default for more than a year now
- These rules were ported from CA rules to IDE rules to enable shipping them in the box. This is no longer required as all CA and IDE rules ship with the .NET SDK now.

Closes #38984
Closes #38944
Closes #38204
Closes #36507
Closes #38900

### CODE REVIEW

File: src/Analyzers/Core/Analyzers/IDEDiagnosticIds.cs [hunk 1]
```diff
@@ -113,9 +113,10 @@ internal static class IDEDiagnosticIds
 
         public const string ConvertSwitchStatementToExpressionDiagnosticId = "IDE0066";
 
-        public const string DisposeObjectsBeforeLosingScopeDiagnosticId = "IDE0067";
-        public const string UseRecommendedDisposePatternDiagnosticId = "IDE0068";
-        public const string DisposableFieldsShouldBeDisposedDiagnosticId = "IDE0069";
+        // IDE0067-IDE0069 deprecated in favor of CA2000 and CA2213
+        // public const string DisposeObjectsBeforeLosingScopeDiagnosticId = "IDE0067";
+        // public const string UseRecommendedDisposePatternDiagnosticId = "IDE0068";
+        // public const string DisposableFieldsShouldBeDisposedDiagnosticId = "IDE0069";
```
Comment (Youssef1313) [inline src/Analyzers/Core/Analyzers/IDEDiagnosticIds.cs:119]: nit: Consider commenting the corresponding entries in `EnforceOnBuildValues`

Comment (mavasani) [reply src/Analyzers/Core/Analyzers/IDEDiagnosticIds.cs:119]: Thanks - I am going to just delete those values

File: src/Features/Core/Portable/DisposeAnalysis/DisposableFieldsShouldBeDisposedDiagnosticAnalyzer.cs [hunk 2]
```diff
@@ -1,333 +0,0 @@
...
-            : base(ImmutableArray.Create(descriptor), GeneratedCodeAnalysisFlags.Analyze)
-        {
-            _disposableFieldsShouldBeDisposedRule = descriptor;
-        }
-
-        private static DiagnosticDescriptor CreateDescriptor(bool isEnabledByDefault)
-            => CreateDescriptor(
-                IDEDiagnosticIds.DisposableFieldsShouldBeDisposedDiagnosticId,
-                EnforceOnBuildValues.DisposableFieldsShouldBeDisposed,
-                title: new LocalizableResourceString(nameof(FeaturesResources.Disposable_fields_should_be_disposed), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
```
Comment (Youssef1313) [inline src/Features/Core/Portable/DisposeAnalysis/DisposableFieldsShouldBeDisposedDiagnosticAnalyzer.cs:50]: Delete corresponding resources as well?

Comment (mavasani) [reply src/Features/Core/Portable/DisposeAnalysis/DisposableFieldsShouldBeDisposedDiagnosticAnalyzer.cs:50]: Ah good call

**Comment by Youssef1313** [comment]
From https://github.com/dotnet/roslyn/issues/38900#issuecomment-536031494:

> 📝 Note that IDE0069 and other diagnostics are not intended to directly duplicate prior CA* warnings. The new rules are reviewed for ongoing correctness against current practices, and may be adjusted in various ways.

This seem to conflict with the reasoning of removal:

> These rules were ported from CA rules to IDE rules to enable shipping them in the box. This is no longer required as all CA and IDE rules ship with the .NET SDK now.

cc: @sharwell

**Comment by mavasani** [comment]
@Youssef1313 The IDE dispose rules were mostly a port from CA rules. It may have had some fine tuning done with time, but they are almost the same. These are code quality rules, and do not belong to the CodeStyle/Unnecessary code rules present in Roslyn repo.

**Review by genlu — APPROVED** [review]
:shipit:

**Review by jmarolf — APPROVED** [review]

**Pull request merged**